### PR TITLE
Allow an external sidebar

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Allow mobile apps to have their own sidebar:

- UI responds to new message `{"type": "notifications/show"}` to show the notification sidebar
- ExternalSettings can contain new `hasSidebar` boolean
- When the menu button is pressed and external sidebar present, fire external message
  ```json
  {
    "type": "sidebar/show"
  }
  ```
- Break up external logic and have a 2nd piece loaded by the app when the UI is set up. This allows us to remove the mixin and some other bookkeeping.
- The UI will now always fetch the external config when loading external auth instead of having each component that uses it fetch their own. It's static. This allowed removing a bunch of 'track external config' code 

CC @zacwest

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zacwest** [comment]
Can you stop this highlighted state from happening when the button is tapped?



I can't get the sidebar button to show up on larger screens, e.g. on iPad:

Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) (15 2) - 2022-01-22 at 18 15 48

**Comment by zacwest** [comment]
Ah right, the sidebar's button is in the sidebar on the larger screen sizes. I can probably preserve that behavior.

**Comment by balloob** [comment]
Worked with Zac on the remaining highlight on the menu button. The ripple of mwc-icon-button remains in a hovered state. At this point unable to figure out what causes it / how we can fix it.

**Comment by balloob** [comment]
Zac confirmed that this PR is ok.

File: src/layouts/home-assistant-main.ts [hunk 1]
```diff
@@ -107,6 +110,14 @@ class HomeAssistantMain extends LitElement {
   protected firstUpdated() {
     import(/* webpackPreload: true */ "../components/ha-sidebar");
 
+    if (this.hass.auth.external) {
+      this._externalSidebar =
+        this.hass.auth.external.config.hasSidebar === true;
+      import("../external_app/external_app_entrypoint").then((mod) =>
```
Comment (balloob) [inline src/layouts/home-assistant-main.ts:116]: All external app stuff that needs to be available after initialization is now loaded here.

**Review by bramkragten — APPROVED** [review]

**Comment by LasseRosenow** [comment]
> Worked with Zac on the remaining highlight on the menu button. The ripple of mwc-icon-button remains in a hovered state. At this point unable to figure out what causes it / how we can fix it.

As far as I know MWC components are supposed to behave like this. They implement a background color for :focus, so they keep a background as long as they are in focus, which is a shame since :focus-visible is s thing, but the last time I reached out to them they were not interested in changing that :(

**Comment by balloob** [comment]
Well I would expect the hovered state to be removed when I call "blur" but that didn't work.

`focus-visible` is only just landing in Safari (next release).

**Pull request merged**